src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
   scene.remove(trajectory);
   /**
-   * The x axis controls the left and right direction, the y axis controls
-   * up and down movement, and the z axis controls forward and back movement.
+   * The X axis controls the left and right direction, the Y axis controls
+   * forward and back movement, and the Z axis controls up and down movement.
    */
-  let coordinates=[];
-  for (point of pose) {
-    coordinates.push(new THREE.Vector3(
-      /**
-       * Subtracting each point by the first point starts the pose
-       * at coordinates (0, 0, 0). Each unit in the 3D scene is 4
-       * meters while the 6th decimal point in GPS coordinates represents
-       * .11 meters. This is why we multiple the lat/lng by 25000, an
-       * increment in the 6th decimal place equates to a 0.025 unit
-       * change in our 3D space. Since altitude is already represented in
-       * meters, we simply divide by 4 to adjust for our 1:4 unit ratio.
-       */
-      (point.lng - pose[0].lng) * 25000 * poseTransform.scale,
-      (point.alt - pose[0].alt)/4,
-      (point.lat - pose[0].lat) * 25000 * -poseTransform.scale)
-    );
+  const coordinates=[];
+  for (const point of pose) {
+    const [x0, y0, z0] = llaToEcef(point.lat, point.lng, point.alt);
+    const [x, y, z] = ecefToEnu(x0, y0, z0, median.lat, median.lng, median.alt);
+    coordinates.push(new THREE.Vector3(x, y, z)
+        .multiplyScalar(poseTransform.scale));
   }
   const geometry = new THREE.BufferGeometry().setFromPoints(coordinates);
-  const material = new THREE.LineBasicMaterial({color: 'blue'});
+  const material = new THREE.LineBasicMaterial({color: 'green'});
   trajectory = new THREE.Line(geometry, material);
   scene.add(trajectory);
+  // Allows the GUI to manipulate the trajectory.
   trajectory.position.x = poseTransform.translateX;
+  trajectory.position.y = poseTransform.translateY;
   trajectory.position.z = poseTransform.translateZ;
-  trajectory.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
+  trajectory.rotation.y = THREE.Math.degToRad(poseTransform.rotateY);
+  trajectory.rotation.z = THREE.Math.degToRad(poseTransform.rotateZ);
 }
 
 /**
- * This repositions all children the orientation object. The object 
- * is a geometry instance, initializing the object with one geometry
- * call rather than the number of data points. (O(1) vs. O(N))
+ * Controls all 3 geometry instances, along with positioning the individual
+ * instances.
  */
 function plotOrientation() {
-  for (point of pose){
+  for (const axis in orientation) {
+    if (axis != '') {
+      matrixRotation(axis);
+      // Allows the GUI to manipulate the orientation.
+      orientation[axis].instanceMatrix.needsUpdate = true;
+      orientation[axis].position.x = poseTransform.translateX;
+      orientation[axis].position.y = poseTransform.translateY;
+      orientation[axis].position.z = poseTransform.translateZ;
+      orientation[axis].rotation.y = THREE.Math.degToRad(poseTransform.rotateY);
```

Comment:
is there no need to rotate on the x axis?

Reply:
Don't see why, any rotation on the x axis usually clips the pose through the 2D map. Also, the ENU coordinate system should always keep the pose parallel to the ground.